### PR TITLE
Added documentation and formatting

### PR DESCRIPTION
**Pull request by jpreece**
Added some documentation to all the custom method and data structures for utilities, profiles, auth, and compute.  I also added a new interface under auth.ts for credentials and wrote a method to support sync prompting of user credentials. Let me know if this is cleaner or just causes more issues?

### CODE REVIEW

**Review by jpreece — COMMENTED** [review]
Think of adding a sanitize method for cleaning up profile after prompt(), this will take a profile and return back the sanitized profile.  This will make that portion easily testable.

File: client/test/profile.test.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    it('get active profile when no profile active', async function () {
+      const activeProfile = await profileConfig.getActiveProfile();
+      expect(activeProfile).to.be.equal(undefined, 'No active profile should be found');
+    });
+
+    it('set active profile', async function () {
+      await profileConfig.setActiveProfile("test");
+      const activeProfile = await profileConfig.getActiveProfile();
+      expect(activeProfile['name']).to.equal('test', 'Active profile has been set to test');
+      expect(activeProfile['profile']['sas-endpoint'], 'https://test-host.sas.com', 'Active profile endpoint not expected')
```
Comment (mattschleder) [inline client/test/profile.test.ts:64]:
```suggestion
      expect(activeProfile['profile']['sas-endpoint']).to.equal('https://test-host.sas.com', 'Active profile endpoint not expected');
```

File: client/test/profile.test.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      // update a profile that already exists
+      expect(testProfile['sas-endpoint']).to.equal('https://test2-host.sas.com');
+      expect(testProfile['client-id']).to.equal('sas.test');
+
+      // update a profile to a new token file
+      testProfile['client-id'] = ''
+      testProfile['token-file'] = '/home/test/.sas/token.txt'
+      profileConfig.upsertProfile("test", testProfile);
+
+      // validate te sanituze update the profile
```
Comment (mattschleder) [inline client/test/profile.test.ts:82]:
```suggestion
      // validate to sanitize profile after profile update
```

**Review by mattschleder — APPROVED** [review]
Approved with a couple of minor tweaks.

**Pull request merged**